Lib/zipfile/__init__.py

```diff
@@ -614,7 +614,11 @@ def _for_archive(self, archive: ZipFile) -> Self:
 
         Return self.
         """
-        self.date_time = time.localtime(time.time())[:6]
+        # gh-91279: Set the SOURCE_DATE_EPOCH to a specific timestamp
+        epoch = os.environ.get('SOURCE_DATE_EPOCH')
+        get_time = int(epoch) if epoch else time.time()
+        self.date_time = time.gmtime(get_time)[:6]
```

Comment:
I'm a little uneasy that `time.localtime()` is now replaced by `time.gmtime()`. That means that the value will change depending on the local time zone of the machine running the code, which may not be adequately exercised in the code. Are there existing tests that validate that `gmtime` is the correct usage here?

Reply:
I checked the documentation. Shouldn't `time.gmtime` return UTC time and not change with time zones?? `time.localtime` says "Like gmtime() but converts to local time"

https://docs.python.org/3.14/library/time.html#time.gmtime

Reply:
What I mean is that this change alters the behavior for all cases. The issue doesn't make mention of gmtime and only mentions localtime in relation to the current implementation. What's the motivation for changing it?

Reply:
Sorry, I see that I was mistaken. To implement this PR I originally thought it was necessary to change it to gmtime, but in fact, this modification should not have been made😥